src/main/groovy/com/palantir/gradle/gitversion/GitVersionPlugin.groovy

```diff
@@ -128,4 +132,31 @@ class GitVersionPlugin implements Plugin<Project> {
         // look in parent directory
         return scanForRootGitDir(currentRoot.parentFile)
     }
+
+    private static String runGitCommand(File dir, String ...commands) {
+        List<String> cmdInput = new ArrayList<>()
+        cmdInput.add("git")
+        cmdInput.addAll(commands)
+        ProcessBuilder pb = new ProcessBuilder(cmdInput)
+        pb.directory(dir)
+        pb.redirectErrorStream(true)
+
+        Process process = pb.start()
+        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))
+
+        StringBuilder builder = new StringBuilder()
+        String line = null
+        while ((line = reader.readLine()) != null) {
+            builder.append(line)
+            builder.append(System.getProperty("line.separator"))
+        }
+
+        int exitCode = process.waitFor()
+        if (exitCode != 0) {
+            return ""
```

Comment:
We can still discuss the broader issue, but added check for `git` executable. If it does not exist, throws an exception of the following form (this is from unit test output when I changed the check to check for an executable named `gitz`, which obviously does not exist):

```
|           * What went wrong:
|           A problem occurred evaluating root project 'junit2262689230640608264'.
|           > Cannot run program "gitz": error=2, No such file or directory
```